spark/src/test/scala/org/apache/spark/sql/delta/InCommitTimestampSuite.scala

```diff
@@ -441,5 +445,401 @@ class InCommitTimestampSuite
       assert(blob("exceptionStackTrace").contains(Snapshot.getClass.getName.stripSuffix("$")))
     }
   }
+
+  test("DeltaHistoryManager.getActiveCommitAtTimeFromICTRange") {
+    withTempDir { tempDir =>
+      spark.range(10).write.format("delta").save(tempDir.getAbsolutePath)
+      val startTime = System.currentTimeMillis()
+      val clock = new ManualClock(startTime)
+      // Ensure that a cached version of deltaLog without the ManualClock is not used.
+      DeltaLog.clearCache()
```

Comment:
That seems... bad... should the clock be part of the cache key??

Reply:
Hmm, if the constructor, which takes a clock as an argument, is mostly used only for testing, it might not be worth maintaining the clock object in the key.